stats/test/platform-routes.test.js

```diff
@@ -79,40 +71,25 @@ describe('Platform Routes HTTP request handler', () => {
       await assertResponseStatus(res, 200)
       const metrics = await res.json()
       assert.deepStrictEqual(metrics, [
-        { day: '2024-01-11', station_id_count: 1 },
-        { day: '2024-01-12', station_id_count: 2 }
+        { day: '2024-01-11', station_count: 3 },
+        { day: '2024-01-12', station_count: 4 }
       ])
     })
   })
 
   describe('GET /stations/monthly', () => {
-    it('returns monthly station metrics for the given date range ignoring the day number', async () => {
+    it('returns monthly station metrics for the given date range', async () => {
       // before the date range
-      await givenDailyStationMetrics(pgPools.evaluate, '2023-12-31', [
-        { ...STATION_STATS, acceptedMeasurementCount: 1, totalMeasurementCount: 1 }
-      ])
+      await givenMonthlyActiveStationCount(pgPools.evaluate, '2023-12-01', 1)
       // in the date range
-      await givenDailyStationMetrics(pgPools.evaluate, '2024-01-10', [
-        { ...STATION_STATS, acceptedMeasurementCount: 1, totalMeasurementCount: 1 }
-      ])
-      await givenDailyStationMetrics(pgPools.evaluate, '2024-01-11', [
-        { ...STATION_STATS, stationId: 'station2', acceptedMeasurementCount: 1, totalMeasurementCount: 1 }
-      ])
-      await givenDailyStationMetrics(pgPools.evaluate, '2024-01-12', [
-        { ...STATION_STATS, stationId: 'station2', acceptedMeasurementCount: 2, totalMeasurementCount: 2 },
-        { ...STATION_STATS, stationId: 'station3', acceptedMeasurementCount: 1, totalMeasurementCount: 1 }
-      ])
-      await givenDailyStationMetrics(pgPools.evaluate, '2024-02-13', [
-        { ...STATION_STATS, acceptedMeasurementCount: 1, totalMeasurementCount: 1 }
-      ])
+      await givenMonthlyActiveStationCount(pgPools.evaluate, '2024-01-01', 3)
+      await givenMonthlyActiveStationCount(pgPools.evaluate, '2024-02-01', 1)
       // after the date range
-      await givenDailyStationMetrics(pgPools.evaluate, '2024-03-01', [
-        { ...STATION_STATS, acceptedMeasurementCount: 1, totalMeasurementCount: 1 }
-      ])
+      await givenMonthlyActiveStationCount(pgPools.evaluate, '2024-03-01', 1)
 
       const res = await fetch(
         new URL(
-          '/stations/monthly?from=2024-01-11&to=2024-02-11',
+          '/stations/monthly?from=2024-01-01&to=2024-02-11',
```

Comment:
❗ I think this is going to break the dashboard, and we need to preserve the original behavior.

In Grafana, we don't have a way to extract the month from the date; we always send the full date string. The REST API drops the day from the date string.

Take a look how `getMonthlyDistinctCount` does it using `date_trunc('month', value)`

```sql
    SELECT
      date_trunc('month', day)::DATE::TEXT as month,
      COUNT(DISTINCT ${safeColumn})::INT as ${safeAsColumn}
    FROM ${safeTable}
    WHERE
      day >= date_trunc('month', $1::DATE)
      AND day < date_trunc('month', $2::DATE) + INTERVAL '1 month'
    GROUP BY month
    ORDER BY month
```

Reply:
I'm sorry, I'm not sure I understand what would break the dashboard here?

The change to the test here was just so that we can test that it can grab both months correctly. If we prefer to do 2023-12-11 to 2024-02-11, I can do that. Not sure if that's what you mean though.

Reply:
I'd like to preserve the behaviour where `/stations/monthly?from=2024-01-11&to=2024-02-11` return data for both January and February.

Can you please revert the change on this line to verify that the new version handles this (edge?) case in the same way as the old version, and then add a new test for `/stations/monthly?from=2024-01-01&to=2024-02-11` if desired.

Reply:
Ah I see... Did not remember that `from=01-11` would also end up showing january data. I made changes to the test and to the query in https://github.com/filecoin-station/spark-stats/pull/210/commits/5362184846bbca36147ce4042a10ef6dcca1be76